pkg/noun/events.c

```diff
@@ -398,23 +398,41 @@ _ce_ephemeral_open(c3_i* eph_i)
 /* _ce_image_open(): open or create image.
 */
 static c3_o
-_ce_image_open(u3e_image* img_u)
+_ce_image_open(u3e_image* img_u, c3_c* ful_c)
 {
   c3_i mod_i = O_RDWR | O_CREAT;
-  c3_c ful_c[8193];
 
-  snprintf(ful_c, 8192, "%s", u3P.dir_c);
-  c3_mkdir(ful_c, 0700);
+  //  create directories if necessary
+  //
+  {
+    c3_c pax_c[8192];
+    c3_c* end_c = pax_c + 8192;
+    c3_c* cur_c = pax_c + snprintf(pax_c, 8192, "%s", ful_c);
 
-  snprintf(ful_c, 8192, "%s/.urb", u3P.dir_c);
-  c3_mkdir(ful_c, 0700);
+    while ( cur_c < end_c ) {
+      c3_c* nxt_c = strchr(cur_c, '/');
 
-  snprintf(ful_c, 8192, "%s/.urb/chk", u3P.dir_c);
-  c3_mkdir(ful_c, 0700);
+      if ( !nxt_c ) {
+        break;
+      }
+      else {
+        *nxt_c = 0;
+
+        if ( 0 != mkdir(pax_c, 0777) && EEXIST != errno ) {
+          fprintf(stderr, "loom: mkdir %s: %s\r\n", pax_c, strerror(errno));
+          return c3n;
+        }
+
+        *nxt_c = '/';
+        cur_c = nxt_c + 1;
+      }
+    }
+  }
```

Comment:
wtf is this? cur_c starts at the *end* of the string and thus this while loop doesn't do anything useful. I guess you're trying to do the equivalent of `mkdir -p /my/new/path` ?

Reply:
moreover, since we're in .urb, I don't think we should be doing this at all

Reply:
Removed this entirely as it is unnecessary, thank you.